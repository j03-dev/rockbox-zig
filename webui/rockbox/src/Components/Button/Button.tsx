import { FC } from "react";
import { Button as BaseButton, KIND } from "baseui/button";

export type ButtonProps = {
  onClick: () => void;
  children: string | JSX.Element;
  kind?: "primary" | "secondary" | "tertiary";
  width?: string;
  height?: string;
  borderRadius?: string;
  disabled?: boolean;
};

const Button: FC<ButtonProps> = ({
  children,
  kind,
  width,
  height,
  borderRadius,
  onClick,
  disabled,
}) => {
  return (
    <BaseButton
      kind={kind}
      overrides={{
        BaseButton: {
          style: ({ $disabled }) => ({
            width,
            height,
            borderTopLeftRadius: borderRadius,
            borderTopRightRadius: borderRadius,
            borderBottomLeftRadius: borderRadius,
            borderBottomRightRadius: borderRadius,
            fontSize: "14px",
            fontFamily: "RockfordSansMedium",
            opacity: $disabled ? 0.7 : 1,
          }),
        },
      }}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </BaseButton>
  );
};

Button.defaultProps = {
  kind: KIND.primary,
  width: "141px",
  height: "40px",
  borderRadius: "20px",
  disabled: false,
};

export default Button;
