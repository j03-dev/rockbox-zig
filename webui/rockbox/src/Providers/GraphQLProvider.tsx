import {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
  ApolloProvider,
  split,
  from,
} from "@apollo/client";
import { removeTypenameFromVariables } from "@apollo/client/link/remove-typename";
import { WebSocketLink } from "@apollo/client/link/ws";
import { getMainDefinition } from "@apollo/client/utilities";
import { SubscriptionClient } from "subscriptions-transport-ws";
import { FC, ReactNode } from "react";

const uri =
  process.env.NODE_ENV === "development"
    ? import.meta.env.VITE_APP_API_URL || "http://localhost:6062/graphql"
    : `${origin}/graphql`;

const removeTypenameLink = removeTypenameFromVariables();

const httpLink = createHttpLink({
  uri,
});

const wsLink = new WebSocketLink(
  new SubscriptionClient(uri.replace("http", "ws"))
);

const splitLink = split(
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === "OperationDefinition" &&
      definition.operation === "subscription"
    );
  },
  wsLink,
  from([removeTypenameLink, httpLink])
);
const link = splitLink;

const client = new ApolloClient({
  link,
  cache: new InMemoryCache(),
});

const GraphQLProvider: FC<{
  children: ReactNode;
}> = ({ children }) => {
  return <ApolloProvider client={client}>{children}</ApolloProvider>;
};

export default GraphQLProvider;
