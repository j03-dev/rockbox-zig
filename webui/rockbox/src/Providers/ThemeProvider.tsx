import { createContext, useState, FC } from "react";
import { ThemeProvider as EmotionThemeProvider } from "@emotion/react";
import {
  BaseUIDarkTheme,
  BaseUILightTheme,
  DarkTheme,
  LightTheme,
} from "../Theme";
import { BaseProvider } from "baseui";
import { ThemeProvider as MaterialThemeProvider } from "@mui/material/styles";
import { createTheme } from "@mui/material/styles";

export type Theme = "light" | "dark";

export type ThemeContextType = {
  theme: Theme;
  setTheme: (theme: Theme) => void;
};

export const ThemeContext = createContext<ThemeContextType>({
  theme: "light",
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  setTheme: (_theme: Theme) => {},
});

const muitheme = createTheme({
  cssVariables: true,
  palette: {
    primary: {
      main: "#fe099c",
    },
  },
});

export type ThemeProviderProps = {
  children: React.ReactNode;
};

const ThemeProvider: FC<ThemeProviderProps> = ({ children }) => {
  const [theme, setTheme] = useState<Theme>("light");
  return (
    <MaterialThemeProvider theme={muitheme}>
      <ThemeContext.Provider value={{ theme, setTheme }}>
        <EmotionThemeProvider theme={theme === "dark" ? DarkTheme : LightTheme}>
          <BaseProvider
            theme={theme === "dark" ? BaseUIDarkTheme : BaseUILightTheme}
          >
            {children}
          </BaseProvider>
        </EmotionThemeProvider>
      </ThemeContext.Provider>
    </MaterialThemeProvider>
  );
};

export default ThemeProvider;
