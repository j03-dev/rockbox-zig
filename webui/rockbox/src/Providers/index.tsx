import { FC } from "react";
import ThemeProvider from "./ThemeProvider";
import { Provider as StyletronProvider } from "styletron-react";
import { Client as Styletron } from "styletron-engine-atomic";
import { RecoilRoot } from "recoil";

const engine = new Styletron();

export type ProvidersProps = {
  children: React.ReactNode;
};

const Providers: FC<ProvidersProps> = ({ children }) => {
  return (
    <RecoilRoot>
      <StyletronProvider value={engine}>
        <ThemeProvider>{children}</ThemeProvider>
      </StyletronProvider>
    </RecoilRoot>
  );
};

export default Providers;
