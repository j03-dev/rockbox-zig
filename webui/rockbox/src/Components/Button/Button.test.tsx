import { render } from "@testing-library/react";
import { vi } from "vitest";
import Button from "./Button";
import Providers from "../../Providers";

describe("Button", () => {
  it("should render", () => {
    const { container } = render(
      <Providers>
        <Button onClick={vi.fn()}>Click me</Button>
      </Providers>
    );
    expect(container).toMatchSnapshot();
  });
});
