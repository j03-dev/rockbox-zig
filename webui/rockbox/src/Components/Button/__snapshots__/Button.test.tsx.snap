// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render 1`] = `
<div>
  <div
    class=""
  >
    <button
      class="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd be bf bg bh bi bj bk bl bm bn"
      data-baseweb="button"
    >
      Click me
    </button>
  </div>
  <div
    class=""
  />
</div>
`;
